lib/widgets/nostr_sign_event_dialog.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:breez_translations/breez_translations_locales.dart';
+import 'package:flutter/material.dart';
+
+import 'package:shared_preferences/shared_preferences.dart';
+
+class NostrSignEventDialog extends StatefulWidget {
+  final Map<String, dynamic> eventData;
+  final String eventContent;
+  final String messageKind;
+  final String appName;
+
+  const NostrSignEventDialog({
+    Key key,
+    this.eventData,
+    this.eventContent,
+    this.messageKind,
+    this.appName,
+  }) : super(key: key);
+
+  @override
+  State<NostrSignEventDialog> createState() => _NostrSignEventDialogState();
+}
+
+class _NostrSignEventDialogState extends State<NostrSignEventDialog> {
+  bool _rememberChoice = false;
+
+  void _handleRememberMe(bool value) async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    prefs.setBool('rememberSignEventChoice', value);
+    setState(() {
+      _rememberChoice = value;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final themeData = Theme.of(context);
+    final texts = context.texts();
+
+    return Theme(
+      data: themeData.copyWith(
+        unselectedWidgetColor: themeData.canvasColor,
+      ),
+      child: AlertDialog(
+        titlePadding: const EdgeInsets.fromLTRB(24.0, 22.0, 0.0, 16.0),
+        title: Text(
+          widget.appName,
+          style: themeData.dialogTheme.titleTextStyle,
+        ),
+        contentPadding: const EdgeInsets.fromLTRB(8.0, 8.0, 8.0, 24.0),
+        content: SingleChildScrollView(
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            children: _getContent(context),
```

Comment:
> We don't really need a helper method for this, the widget in `_getContent` can be moved here or extracted as a widget, `_NostrGetPublicKeyDialogContent`.
> 
> We're trying to avoid and replace existing helper methods in our codebase with Widgets where possible. Explanation [here](https://www.youtube.com/watch?v=IOyq-eTRhvo).


Same here.